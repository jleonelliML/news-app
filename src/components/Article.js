import React from 'react';
import Card from 'react-bootstrap/Card';
import { MDBBtn } from "mdbreact";
import ProgressiveImage from "./ProgressiveImage";
import previewImage from "../assets/preview.jpeg"

class Article extends React.Component {
    render() {
        const {author, content, description, title, url, publishedAt, urlToImage} = this.props.article;
        return (
                <Card className="article-content">
                        <ProgressiveImage preview={previewImage} image={urlToImage} />
                    <Card.Body>
                        <Card.Title>{title}</Card.Title>
                        <Card.Text>
                        {content}
                        </Card.Text>
                        <blockquote className="blockquote mb-0">
                            <footer className="blockquote-footer">
                                Author: <cite title="Author">{author}</cite>
                            </footer>
                        </blockquote>
                        <MDBBtn href={url} size="md" className="nav-bar">
                            Read more
                        </MDBBtn>
                    </Card.Body>
                </Card>
        )
    }
}

export default Article;