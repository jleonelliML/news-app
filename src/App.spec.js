import React from 'react';
import renderer from 'react-test-renderer';

import Article from './components/Article';

describe('Counter', () => {
  test('snapshot renders', () => {
    const article = {
        author: "Test"
    }

    const component = renderer.create(<Article article={article}/>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});