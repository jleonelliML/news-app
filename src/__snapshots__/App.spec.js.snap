// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter snapshot renders 1`] = `
<div
  className="article-content card"
>
  <img
    className="card-img-top"
  />
  <div
    className="card-body"
  >
    <div
      className="card-title h5"
    />
    <p
      className="card-text"
    />
    <blockquote
      className="blockquote mb-0"
    >
      <footer
        className="blockquote-footer"
      >
        Author: 
        <cite
          title="Author"
        >
          Test
        </cite>
      </footer>
    </blockquote>
    <button
      className="btn btn-default btn-md Ripple-parent nav-bar"
      onMouseUp={[Function]}
      onTouchStart={[Function]}
      type="button"
    >
      Read more
      <div
        className="Ripple "
        style={
          Object {
            "height": "0px",
            "left": "0px",
            "top": "0px",
            "width": "0px",
          }
        }
      />
    </button>
  </div>
</div>
`;
